File: src/test/java/de/schramm/royalbash/tdd/core/card/NoOpCardTest.java
package de.schramm.royalbash.tdd.core.card;

import de.schramm.royalbash.tdd.core.Context;
import de.schramm.royalbash.tdd.core.Game;
import de.schramm.royalbash.tdd.core.Player;
import lombok.val;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class NoOpCardTest {

    @Test
    public void should_not_alter_game() {

        // Given
        val testee = NoOpCard.builder()
                .build();
        val player1 = Player.builder()
                .handcard(testee)
                .build();
        val player2 = Player.builder().build();
        val game = Game.builder()
                .player1(player1)
                .player2(player2)
                .build();
        val context = Context.builder()
                .game(game)
                .targetPlayer(player2)
                .build();

        // When
        val updatedGame = testee.invoke(context);

        // Then
        assertThat(updatedGame).isEqualTo(game);
    }

}

File: backend/src/test/kotlin/de/schramm/royalbash/domain/card/NoOpCardTest.kt
package de.schramm.royalbash.domain.card

import de.schramm.royalbash.domain.Context
import de.schramm.royalbash.domain.Game
import de.schramm.royalbash.domain.Player
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class NoOpCardTest {

    @Test
    fun should_not_alter_game() {

        // Given
        val testee = CardMock(id = "Id 1", instanceId = "InstanceId 1")
        val player1 = Player("Id 2", handcards = listOf(testee))
        val player2 = Player("Id 3")
        val game = Game("Id 4", player1 = player1, player2 = player2)
        val context = Context(game, player1, player2)

        // When
        val updatedGame = testee.invoke(context)

        // Then
        assertThat(updatedGame).isEqualTo(game)
    }

}
